projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -311,6 +370,29 @@ protected void Enqueue(IRpcContinuation k)
             }
         }
 
+        internal async ValueTask<IChannel> OpenAsync()
+        {
+            await _rpcSemaphore.WaitAsync()
+                .ConfigureAwait(false);
+            using var k = new ChannelOpenAsyncRpcContinuation(ContinuationTimeout);
+            try
+            {
+                Enqueue(k);
+
+                var method = new ChannelOpen();
+                await ModelSendAsync(method)
+                    .ConfigureAwait(false);
+
+                bool result = await k;
```

Comment:
k.ConfigureAwait(false);
There are 2-3 more places for that, you might want to search/replace in this file for it.

Reply:
NetAnalyzers will help with that.

Reply:
In this particular case, `k.ConfigureAwait(false)` is not a valid method call at that point.

The `ChannelOpenAsyncRpcContinuation` already configures the awaitable `Task` here - https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/lukebakken/async-operations/projects/RabbitMQ.Client/client/impl/AsyncRpcContinuations.cs#L72

If that is not the correct way to do it, please follow-up in issue #1269 or open a dedicated issue for it. Thanks!

Reply:
> In this particular case, `k.ConfigureAwait(false)` is not a valid method call on line 386.
> 
> The `ChannelOpenAsyncRpcContinuation` already configures the awaitable `Task` here - https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/lukebakken/async-operations/projects/RabbitMQ.Client/client/impl/AsyncRpcContinuations.cs#L72
> 
> If that is not the correct way to do it, please follow-up in issue #1269 or open a dedicated issue for it. Thanks!

I'm not sure I understand your comment, @lukebakken. Beware that `ConfigureAwait` configures the await, not the task. if the task was already completed, the continuations for subsequent awaits will be scheduled to the captured synchronization context or task scheduler.